### PR TITLE
feat: prevent adding unsupported hooks for your wallet

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Prevents the user from being able to add unsupported hooks for their current connected wallet
- Visually indicates which hooks are not compatible
- Sorts the hooks list automatically by supported first -> then unsupported
- Hooks without specific support defined (n/a) are considered 'compatible'

**EOA connected:**
Screenshot 2024-10-22 at 16 52 30

**Safe connected:**
 
https://github.com/user-attachments/assets/ae523d84-4e26-4038-a8d2-cece770f1152

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/hooksStore/pure/HookDetailHeader/index.tsx [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
   iconSize?: number
   gap?: number
   padding?: string
 }
 
-export function HookDetailHeader({ dapp, onSelect, iconSize, gap, padding }: HookDetailHeaderProps) {
+export function HookDetailHeader({ dapp, walletType, onSelect, iconSize, gap, padding }: HookDetailHeaderProps) {
   const { name, image, descriptionShort } = dapp
 
+  const isCompatible =
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/hooksStore/pure/HookDetailHeader/index.tsx:19]: The code is duplicated in `HookListItem` component, I suggest to move it in a util function

File: apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx [hunk 2]
```diff
@@ -153,13 +134,16 @@ export function HookRegistryList({ onDismiss, isPreHook, hookToEdit }: HookStore
...
-      {filteredDapps.length > 0 ? (
+      {sortedFilteredDapps.length > 0 ? (
         <HookDappsList>
-          {filteredDapps.map((dapp) => (
+          {sortedFilteredDapps.map((dapp) => (
             <HookListItem
               key={isHookDappIframe(dapp) ? dapp.url : dapp.name}
               dapp={dapp}
-              onRemove={isAllHooksTab ? undefined : () => removeCustomHookDapp(dapp as HookDappIframe)}
+              walletType={
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx:143]: `walletType={walletType}`

File: apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx [hunk 3]
```diff
@@ -33,62 +34,48 @@ interface HookStoreModal {
...
+    return currentDapps.filter(({ name = '', descriptionShort = '' }) =>
+      [name, descriptionShort].some((text) => text.toLowerCase().includes(lowerQuery)),
+    )
   }, [currentDapps, searchQuery])
 
+  const sortedFilteredDapps = useMemo(() => {
+    const isCompatible = (dapp: HookDapp) =>
+      !dapp.conditions?.walletCompatibility || dapp.conditions.walletCompatibility.includes(walletType)
+    return filteredDapps.sort((a, b) => (isCompatible(a) === isCompatible(b) ? 0 : isCompatible(a) ? -1 : 1))
+  }, [filteredDapps, isSmartContractWallet])
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx:67]: `walletType` is not present in `useMemo()` deps

Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooksStore/containers/HookRegistryList/index.tsx:67]: And `isSmartContractWallet` is not needed there

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @fairlighteth , great, thank you!
However, in this PR it is impossible to add a custom hook even if a wallet type is allowed for it. 
impossible to add

Please see #4910 : you can find there instructions how to get a custom hook link.

**Comment by fairlighteth** [comment]
@elena-zh address your comment, should work as expected now:
Screenshot 2024-10-23 at 16 13 49
Screenshot 2024-10-23 at 16 13 37
Screenshot 2024-10-23 at 16 12 47

**Comment by fairlighteth** [comment]
@shoom3301 addressed your comments and did some refactoring in general.

**Review by elena-zh — APPROVED** [review]
Works now, thank you!

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**